lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -381,6 +393,88 @@ def stop_event_loop(self):
     stop_event_loop.__doc__=FigureCanvasBase.stop_event_loop_default.__doc__
 
 
+_flow = [Gtk.Orientation.HORIZONTAL, Gtk.Orientation.VERTICAL]
+flow_types = ['horizontal', 'vertical']
+
+
+class WindowGTK3(WindowBase, Gtk.Window):
+    def __init__(self, title, **kwargs):
+        super(WindowGTK3, self).__init__(title=title, **kwargs)
+        self.set_window_title(title)
+
+        try:
+            self.set_icon_from_file(window_icon)
+        except (SystemExit, KeyboardInterrupt):
+            # re-raise exit type Exceptions
+            raise
+        except:
+            # some versions of gtk throw a glib.GError but not
+            # all, so I am not sure how to catch it.  I am unhappy
+            # doing a blanket catch here, but am not sure what a
+            # better way is - JDH
+            verbose.report('Could not load matplotlib icon: %s' % sys.exc_info()[1])
+
+        self._layout = {}
+        self._setup_box('_outer', Gtk.Orientation.VERTICAL, False, None)
+        self._setup_box('north', Gtk.Orientation.VERTICAL, False, '_outer')
```

Comment:
Meh, the first programming language I learnt, Java, used this nomenclature, so does jquery (for websites, i.e. ``WebAgg``), and also ``Tk``.  I have seen ``Qt`` use this nomenclature for tabs.

I don't mind which nomenclature, just so long as we stay clear of Parkinson's law :wink:.

Also just found this (you don't need to look at it, but just leaving it here for future reference) which should help with Qt http://ftp.ics.uci.edu/pub/centos0/ics-custom-build/BUILD/PyQt-x11-gpl-4.7.2/examples/layouts/borderlayout.py